src/contracts/aludel/AludelV3Lib.sol

```diff
@@ -244,4 +246,25 @@ library AludelV3Lib {
         vault.totalStake += amount;
         aludel.totalStake += amount;
     }
+
+    function cleanupExpiredRewardSchedules(
+        IAludelV3.AludelData storage aludel
+    ) internal {
+        // check : measure how much gas this actually saves
+        uint256 length = aludel.rewardSchedules.length;
+        for (uint index = 0; index < length; index++) {
+            IAludelV3.RewardSchedule memory schedule = aludel.rewardSchedules[index];
+
+            // check if the period is already expired
+            if (block.timestamp - schedule.start >= schedule.duration) {
+                // length decreces
```

Comment:
https://www.learncpp.com/cpp-tutorial/comments/

these comments state that which is obvious from looking at the code, they're better suited to explain _why_ the code does what it does